lib/Doctrine/Annotations/Type/ConstantType.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Annotations\Type;
+
+/**
+ * @internal
+ */
+interface ConstantType extends Type
+{
+    /**
+     * @return mixed
+     */
+    public function getValue();
```

Comment:
This confuses me: a constant "has" a type, but "isn't" a type. There's some confusion about the naming of these interfaces, IMO

Reply:
Is `ConstantType` a restriction over possible types? If so, I'd say that an `EnumType` should probably be represented too (as a `ConstantType[]` collection)

Reply:
Constant type is a specialization of generic type, e.g. `5` is ConstantIntegerType which is still IntegerType.
There is no EnumType currently - enum is simply interpreted as UnionType of multiple ConstantTypes.